endpoints.md

```diff
@@ -143,18 +158,51 @@ Get info of project with ID of project
 
 Get all filters that users can use to filter search
 
-## GET `/projects?filter1=value1&filter2=value2`
+### Result 
+
+If successful, results in `200` status code and an an object containing the filters associated with all existing projects. Otherwise, results in a `400` or `500` error status code with a message about the error.
+
+The returned object contains two properties, `skills` and `tags`. Each property's value is an array of objects that have a respective `skill` or `tag` property containing the name of the filter, and a `count` property containing the number of existing projects associated with the filter.
+
+## GET `/projects?skill[]=value1&tag[]=value2&searchQuery=value3`
 
 ### Description
 
-Get projects by searching with filters
+Get projects by searching with filters and keywords
+
+### Usage
+
+Multiple filters can be specified with repetition of query parameters. For example: 
+
+```
+/projects?skill[]=c%2B%2B&skill[]=python&tag[]=food&searchQuery=restaurant+reviews
+```
+### Result
+
+If successful, results in `200` status code and an array containing objects with the info of all projects matching the filtering criteria in the form of key-value pairs. Otherwise, results in a `500` error status code with a message about the error.
 
-## GET `/projects/suggestions/:id`
+Returned projects are returned in the same format as those returned by the [GET /projects endpoint](#get-projects).
+
+Returned projects are guaranteed to be associated with all specified skills and tags. If a `searchQuery` value is provided, returned projects are also guaranteed to contain at least one of the search terms in the project name.
+
+## GET `/projects/suggestions`
 
 ### Description
 
 Get project suggestions for a user
 
+### Header
+
+`authorization`: token (if exists)
+
+### Result
+
+If successful, results in `200` status code and an array containing objects with the info of projects matching the user's skills and interests. Otherwise, results in a `401` or `500` error status code with a message about the error.
+
+Returned projects are returned in the same format as those returned by the [GET /projects endpoint](#get-projects).
+
+Returned projects are guaranteed to be associated with at least one of the user's skills or interests.
```

Comment:
[Urgent] Do you think we should also send back the proportion of user's skills and interests matched with the projects' skills? For example, if a user has HTML, CSS, and React skills, and a project requires HTML and CSS, then there is a 2/2 match. As another example, if a user has HTML as their only skill, and a project requires HTML and CSS, then there is a 1/2 match.